project_cost_report/report.py

```diff
@@ -0,0 +1,436 @@
+
+    If a report category has no analytic line, it is not shown.
+    """
+
+    _name = 'project.cost.report'
+    _description = 'Project Cost Report'
+
+    def _get_rendering_variables(self, project, report_context):
+        """Get the variables used for rendering the qweb report.
+
+        :param project: the project.project record
+        :param report_context: the rendering context
+        :rtype: dict
+        """
+        return {
+            'project': project,
+            'currency': project.company_id.currency_id,
+        }
+
+    def get_project_from_report_context(self, report_context):
+        """Get the project record from the given report context.
+
+        :param report_context: the rendering context
+        :return: a project.project singleton
+        """
+        project_id = report_context.get('active_id')
+
+        if not project_id:
+            raise ValidationError(_(
+                'The cost report was triggered without a project ID in context.'
+            ))
+
+        return self.env['project.project'].browse(project_id)
+
+    @api.model
+    def get_html(self, report_context):
+        """Get the report html given the report context.
+
+        :param dict report_context: the report context.
+        :rtype: bytes
+        """
+        project = self.get_project_from_report_context(report_context)
+        rendering_variables = self._get_rendering_variables(project, report_context)
+        return self.env.ref('project_cost_report.cost_report_html').render(rendering_variables)
+
+    @api.model
+    def get_pdf(self, report_context):
+        """Get the report html given the report context.
+
+        :param dict report_context: the report context.
+        :rtype: bytes
+        """
+        project = self.get_project_from_report_context(report_context)
+        rendering_variables = self._get_rendering_variables(project, report_context)
+        base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        rendering_variables.update({
+            'mode': 'print',
+            'base_url': base_url,
+        })
+        body = self.env['ir.ui.view'].render_template(
+            "project_cost_report.cost_report_pdf",
+            values=rendering_variables,
+        )
+        header = self.env['ir.actions.report'].render_template(
+            "web.internal_layout", values=rendering_variables
+        )
+        header = self.env['ir.actions.report'].render_template(
+            "web.minimal_layout", values=dict(rendering_variables, subst=True, body=header)
+        )
+        return self.env['ir.actions.report']._run_wkhtmltopdf(
+            [body],
+            header=header,
+            landscape=True,
+            specific_paperformat_args={
+                'data-report-margin-top': 10,
+                'data-report-header-spacing': 10
+            }
+        )
+
+    @api.model
+    def get_foldable_categories(self, project_id):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)